src/meta.jl

```diff
@@ -0,0 +1,238 @@
+    return polymake_app
+end
+
+############### macro helpers
+
+function parse_function_call(expr)
+    @assert expr.head == :call
+    func = expr.args[1] # called function
+    args = expr.args[2:end] # its arguments
+
+    templates = Symbol[]
+    # grab template parameters if present
+    if func.head == :curly
+        templates = func.args[2:end]
+        func = func.args[1]
+    end
+
+    @assert func.head == Symbol('.')
+    module_name = func.args[1]
+    func_name = func.args[2].value
+
+    kwargs = Expr[]
+    kwar_idx = findfirst(a -> a isa Expr && a.head == :kw, args)
+    if kwar_idx != nothing
+        kwargs = args[kwar_idx:end]
+        args = args[1:kwar_idx-1]
+    end
+
+    return module_name, func_name, templates, args, kwargs
+end
+
+############### json parsing
+
+########## structs
+
+abstract type PolymakeCallable end
+
+struct PolymakeFunction <: PolymakeCallable
+    jl_function::Symbol
+    pm_name::String
+    app_name::String
+    json::Dict{String, Any}
+
+    PolymakeFunction(pm_name::String, app_name::String) =
+        new(Symbol(pm_name), pm_name, app_name)
+    PolymakeFunction(jl_name::Symbol, pm_name::String, app_name::String) =
+        new(jl_name, pm_name, app_name)
+    PolymakeFunction(jl_name::Symbol, pm_name::String, app_name::String, json_dict) =
+        new(jl_name, pm_name, app_name, json_dict)
+end
+
+struct PolymakeMethod <: PolymakeCallable
+    jl_function::Symbol
+    pm_name::String
+    app_name::String
+    json::Dict{String, Any}
+
+    PolymakeMethod(pm_name::String) =
+        new(Symbol(pm_name), pm_name)
+    PolymakeMethod(jl_name::Symbol, pm_name::String) =
+        new(jl_name, pm_name)
+    PolymakeMethod(jl_name::Symbol, pm_name::String, app_name, json_dict) =
+        new(jl_name, pm_name, app_name, json_dict)
+end
+
+struct PolymakeApp
+    jl_module::Symbol
+    pm_name::String
+    callables::Vector{PolymakeCallable}
+end
+
+########## constructors
+
+function PolymakeCallable(app_name::String, dict::Dict{String, Any},
+    polymake_name=dict["name"], julia_name=Symbol(polymake_name))
+
+    if haskey(dict, "method_of")
+        return PolymakeMethod(julia_name, polymake_name, app_name, dict)
+    else
+        return PolymakeFunction(julia_name, polymake_name, app_name, dict)
+    end
+end
+
+function PolymakeApp(app_json::Dict{String, Any})
+    app_name = app_json["app"]
+    jl_module = appname_module_dict[Symbol(app_json["app"])]
+
+    for f in app_json["functions"]
+        if !haskey(f, "name")
+            @warn "No 'name' field in json node. Malformed json for application '$app_name'?\n node = $f"
+        end
+    end
+
+    callables = [PolymakeCallable(app_name,f) for f in app_json["functions"] if haskey(f, "name")]
+
+    callables = unique(pc -> pm_name(pc), callables)
+
+    return PolymakeApp(jl_module, app_name, callables)
+end
+
+function PolymakeApp(module_name::Symbol, json_file::String, module_file::String)
```

Comment:
This method seems a bit weird. Both `module_file` and `module_name` seem to be unused. Is that on purpose?